### DISCUSSION TITLE
Is it possible to remove default activities?

### DISCUSSION THREAD

**Discussion by tomy2105** [elsa-workflows/elsa-core#802] [Q&A] [answered]
How does one remove activities that are added when AddElsa without any special options is invoked?

**Comment by sfmskywalker** [comment]
The easiest way is to do `AddElsaCore()` and then add the activities you want to be available manually. Basically copy the implementation of `AddElsa` into your own startup class and make changes as necessary.

**Comment by tomy2105** [comment, marked as answer]
Actually core activities are added inside  `AddElsaCore()` in `AddCoreActivities()` so I cannot invoke it, I must reimplement it, along with `AddElsa()`. In addition, `AddCoreActivities()` is invoked after configure action is invoked so I cannot remove default activities inside configure action.

Since `AddElsaCore()` contains other parts I'd rather not duplicate all of it. Would you consider a pull request with a boolean flag added to `ElsaOptions` which would skip adding core activities in `AddCoreActivities()`?

In essence I could then do:
`

            services

                .AddElsa(options => options

                    .NoCoreActivities()   // method to activate skipping of AddCoreActivities()

                    .UseEntityFrameworkPersistence(ef => ef.UseSqlite())

                    ..... other options ......

                );

`

**Reply by sfmskywalker** [reply]
Thanks to your PR, we can now do exactly as you coded above:

`services.AddElsa(options => options.NoCoreActivities())`
